<br><br><right><h1 style="font-size:4em;color:#2467C0">DAY 9 - TOP WOMEN CHESS PLAYERS ANALYSIS USING THE TOP WOMEN CHESS PLAYERS DATASET</h1></right>
<br>
<table>
<col width="550">
<col width="450">
<tr>
This dataset contains information about the top women chess players around the world.

In [1]:
import pandas as pd
import numpy as np

In [64]:
df = pd.read_csv(r"C:\Users\seyi\Desktop\MY DATA SCIENCE WORKS, TUTORIAL AND JOURNEY\100 days of DS\Resources\Projects\Data Science Analysis\Dataset\top_women_chess_players_aug_2020.csv")
df1 = df.copy()

In [65]:
df

Fide id                     Name Federation Gender  Year_of_birth Title  \
0      700070            Polgar, Judit        HUN      F         1976.0    GM   
1     8602980               Hou, Yifan        CHN      F         1994.0    GM   
2     5008123            Koneru, Humpy        IND      F         1987.0    GM   
3     4147103  Goryachkina, Aleksandra        RUS      F         1998.0    GM   
4      700088            Polgar, Susan        HUN      F         1969.0    GM   
...       ...                      ...        ...    ...            ...   ...   
8548  3302288        Reinkens, Natalia        BOL      F            NaN   NaN   
8549   343960        Saffova, Michaela        CZE      F         1994.0   NaN   
8550  5038294         Shetye, Siddhali        IND      F         1992.0   NaN   
8551  2072491            Trakru, Priya        USA      F         2001.0   WFM   
8552  4666399      Vorpahl, Sina Fleur        GER      F         1982.0   NaN   

      Standard_Rating  Rapid_rating  Blitz_rating Inactive_flag  
0                2675        2646.0        2736.0            wi  
1                2658        2621.0        2601.0           NaN  
2                2586        2483.0        2483.0           NaN  
3                2582        2502.0        2441.0           NaN  
4                2577           NaN           NaN            wi  
...               ...           ...           ...           ...  
8548             1801           NaN           NaN            wi  
8549             1801        1791.0        1765.0           NaN  
8550             1801        1884.0        1824.0            wi  
8551             1801           NaN           NaN            wi  
8552             1801           NaN           NaN            wi  

[8553 rows x 10 columns]

# Data Cleaning

In [66]:
df.isnull()

Fide id   Name  Federation  Gender  Year_of_birth  Title  \
0       False  False       False   False          False  False   
1       False  False       False   False          False  False   
2       False  False       False   False          False  False   
3       False  False       False   False          False  False   
4       False  False       False   False          False  False   
...       ...    ...         ...     ...            ...    ...   
8548    False  False       False   False           True   True   
8549    False  False       False   False          False   True   
8550    False  False       False   False          False   True   
8551    False  False       False   False          False  False   
8552    False  False       False   False          False   True   

      Standard_Rating  Rapid_rating  Blitz_rating  Inactive_flag  
0               False         False         False          False  
1               False         False         False           True  
2               False         False         False           True  
3               False         False         False           True  
4               False          True          True          False  
...               ...           ...           ...            ...  
8548            False          True          True          False  
8549            False         False         False           True  
8550            False         False         False          False  
8551            False          True          True          False  
8552            False          True          True          False  

[8553 rows x 10 columns]

In [67]:
df.isnull().sum()

Fide id               0
Name                  0
Federation            0
Gender                0
Year_of_birth       292
Title              5435
Standard_Rating       0
Rapid_rating       4945
Blitz_rating       5081
Inactive_flag      2701
dtype: int64

In [68]:
df['Year_of_birth'] = df['Year_of_birth'].fillna(method = 'ffill')

In [69]:
df['Year_of_birth'].isnull().any()

False

In [70]:
df['Title'] = df['Title'].fillna(method = 'ffill')

In [71]:
df['Title'].isnull().any()

False

In [72]:
df['Rapid_rating'] = df['Rapid_rating'].fillna(method = 'ffill')

In [73]:
df['Rapid_rating'].isnull().any()

False

In [74]:
df['Blitz_rating'] = df['Blitz_rating'].fillna(method = 'ffill')

In [75]:
df['Blitz_rating'].isnull().any()

False

In [76]:
df['Inactive_flag'] = df['Inactive_flag'].fillna(method = 'ffill')

In [77]:
df['Inactive_flag'].isnull().any()

False

In [78]:
df.isnull().sum()

Fide id            0
Name               0
Federation         0
Gender             0
Year_of_birth      0
Title              0
Standard_Rating    0
Rapid_rating       0
Blitz_rating       0
Inactive_flag      0
dtype: int64

In [79]:
df.head()

Fide id                     Name Federation Gender  Year_of_birth Title  \
0   700070            Polgar, Judit        HUN      F         1976.0    GM   
1  8602980               Hou, Yifan        CHN      F         1994.0    GM   
2  5008123            Koneru, Humpy        IND      F         1987.0    GM   
3  4147103  Goryachkina, Aleksandra        RUS      F         1998.0    GM   
4   700088            Polgar, Susan        HUN      F         1969.0    GM   

   Standard_Rating  Rapid_rating  Blitz_rating Inactive_flag  
0             2675        2646.0        2736.0            wi  
1             2658        2621.0        2601.0            wi  
2             2586        2483.0        2483.0            wi  
3             2582        2502.0        2441.0            wi  
4             2577        2502.0        2441.0            wi

In [80]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8553 entries, 0 to 8552
Data columns (total 10 columns):
Fide id            8553 non-null int64
Name               8553 non-null object
Federation         8553 non-null object
Gender             8553 non-null object
Year_of_birth      8553 non-null float64
Title              8553 non-null object
Standard_Rating    8553 non-null int64
Rapid_rating       8553 non-null float64
Blitz_rating       8553 non-null float64
Inactive_flag      8553 non-null object
dtypes: float64(3), int64(2), object(5)
memory usage: 668.3+ KB


In [81]:
df.shape

(8553, 10)

In [82]:
df.columns

Index(['Fide id', 'Name', 'Federation', 'Gender', 'Year_of_birth', 'Title',
       'Standard_Rating', 'Rapid_rating', 'Blitz_rating', 'Inactive_flag'],
      dtype='object')

In [83]:
df.describe()

Fide id  Year_of_birth  Standard_Rating  Rapid_rating  \
count  8.553000e+03    8553.000000      8553.000000   8553.000000   
mean   8.829011e+06    1985.087805      2005.102303   1925.200164   
std    9.226777e+06      14.164389       137.146646    175.878431   
min    1.001450e+05    1920.000000      1801.000000   1224.000000   
25%    2.119447e+06    1978.000000      1891.000000   1814.000000   
50%    4.500539e+06    1988.000000      1998.000000   1924.000000   
75%    1.360553e+07    1995.000000      2090.000000   2040.000000   
max    7.360114e+07    2010.000000      2675.000000   2646.000000   

       Blitz_rating  
count   8553.000000  
mean    1915.207296  
std      172.655611  
min     1159.000000  
25%     1810.000000  
50%     1917.000000  
75%     2026.000000  
max     2736.000000

# Show the value counts of the various Federations

In [84]:
df['Federation'].value_counts()

RUS    1698
GER     621
POL     384
UKR     287
IND     284
       ... 
CPV       1
QAT       1
NAM       1
BER       1
MAD       1
Name: Federation, Length: 124, dtype: int64

# Top 10 Women Chess Players

In [85]:
top_10_players = df.head(10)
top_10_players

Fide id                     Name Federation Gender  Year_of_birth Title  \
0    700070            Polgar, Judit        HUN      F         1976.0    GM   
1   8602980               Hou, Yifan        CHN      F         1994.0    GM   
2   5008123            Koneru, Humpy        IND      F         1987.0    GM   
3   4147103  Goryachkina, Aleksandra        RUS      F         1998.0    GM   
4    700088            Polgar, Susan        HUN      F         1969.0    GM   
5   8600147                 Xie, Jun        CHN      F         1970.0    GM   
6   8603006               Ju, Wenjun        CHN      F         1991.0    GM   
7  14109336          Lagno, Kateryna        RUS      F         1989.0    GM   
8  14114550         Muzychuk, Mariya        UKR      F         1992.0    GM   
9  12801259       Cmilyte, Viktorija        LTU      F         1983.0    GM   

   Standard_Rating  Rapid_rating  Blitz_rating Inactive_flag  
0             2675        2646.0        2736.0            wi  
1             2658        2621.0        2601.0            wi  
2             2586        2483.0        2483.0            wi  
3             2582        2502.0        2441.0            wi  
4             2577        2502.0        2441.0            wi  
5             2574        2502.0        2441.0            wi  
6             2560        2610.0        2536.0            wi  
7             2546        2521.0        2608.0            wi  
8             2544        2506.0        2326.0            wi  
9             2538        2394.0        2435.0            wi

# Top 10 Women Players and their various ratings and federations

In [86]:
top_players = df[['Name', 'Federation', 'Standard_Rating', 'Rapid_rating', 'Blitz_rating']].groupby(['Name']).head(10)

In [87]:
top_players.head(10)

Name Federation  Standard_Rating  Rapid_rating  \
0            Polgar, Judit        HUN             2675        2646.0   
1               Hou, Yifan        CHN             2658        2621.0   
2            Koneru, Humpy        IND             2586        2483.0   
3  Goryachkina, Aleksandra        RUS             2582        2502.0   
4            Polgar, Susan        HUN             2577        2502.0   
5                 Xie, Jun        CHN             2574        2502.0   
6               Ju, Wenjun        CHN             2560        2610.0   
7          Lagno, Kateryna        RUS             2546        2521.0   
8         Muzychuk, Mariya        UKR             2544        2506.0   
9       Cmilyte, Viktorija        LTU             2538        2394.0   

   Blitz_rating  
0        2736.0  
1        2601.0  
2        2483.0  
3        2441.0  
4        2441.0  
5        2441.0  
6        2536.0  
7        2608.0  
8        2326.0  
9        2435.0

* **Judit Plogar** is the highest rated female player with Standard Rating of 2675. 
* **Hou Yifan** is the highest rated active female player with Standard Rating of 2658.

# Show the value counts of the various titles

Before cleaning data

In [88]:
df1['Title'].value_counts()

WFM    1545
WIM     809
WGM     316
WCM     247
IM      119
GM       37
FM       36
CM        8
WH        1
Name: Title, dtype: int64

After cleaning data

In [89]:
df['Title'].value_counts()

WFM    4733
WIM    1666
WCM    1525
WGM     384
IM      125
FM       66
GM       37
CM       16
WH        1
Name: Title, dtype: int64

# Who are the Chess GrandMasters (GM)

In [90]:
df[df['Title'] == 'GM']

Fide id                      Name Federation Gender  Year_of_birth  \
0      700070             Polgar, Judit        HUN      F         1976.0   
1     8602980                Hou, Yifan        CHN      F         1994.0   
2     5008123             Koneru, Humpy        IND      F         1987.0   
3     4147103   Goryachkina, Aleksandra        RUS      F         1998.0   
4      700088             Polgar, Susan        HUN      F         1969.0   
5     8600147                  Xie, Jun        CHN      F         1970.0   
6     8603006                Ju, Wenjun        CHN      F         1991.0   
7    14109336           Lagno, Kateryna        RUS      F         1989.0   
8    14114550          Muzychuk, Mariya        UKR      F         1992.0   
9    12801259        Cmilyte, Viktorija        LTU      F         1983.0   
10   14111330            Muzychuk, Anna        UKR      F         1990.0   
11   13601903           Dzagnidze, Nana        GEO      F         1987.0   
12    5015197        Harika, Dronavalli        IND      F         1991.0   
13    8603642              Tan, Zhongyi        CHN      F         1991.0   
14    8605114              Lei, Tingjie        CHN      F         1997.0   
15   13600036       Chiburdanidze, Maia        GEO      F         1961.0   
20    8601283                 Zhao, Xue        CHN      F         1985.0   
22    4134974      Kosintseva, Nadezhda        RUS      F         1985.0   
26    4133471       Kosintseva, Tatiana        RUS      F         1986.0   
29    4128125      Kosteniuk, Alexandra        RUS      F         1984.0   
32    8600635                 Xu, Yuhua        CHN      F         1976.0   
33    1700030             Cramling, Pia        SWE      F         1963.0   
34    2902257      Stefanova, Antoaneta        BUL      F         1979.0   
36   13602640       Khotenashvili, Bela        GEO      F         1988.0   
37   13602993        Batsiashvili, Nino        GEO      F         1987.0   
46     617822              Sebag, Marie        FRA      F         1986.0   
49    4167570         Gunina, Valentina        RUS      F         1989.0   
50    2012782              Krush, Irina        USA      F         1983.0   
53    1106619             Socko, Monika        POL      F         1978.0   
56    8600546                 Zhu, Chen        QAT      F         1976.0   
65   14110911            Ushenina, Anna        UKR      F         1985.0   
72   13300210          Danielian, Elina        ARM      F         1978.0   
73   12400149        Hoang, Thanh Trang        HUN      F         1980.0   
101  13600168  Arakhamia-Grant, Ketevan        SCO      F         1968.0   
130   1004786             Peng, Zhaoqin        NED      F         1968.0   
180  14101513          Zhukova, Natalia        UKR      F         1979.0   
350  13600125      Gaprindashvili, Nona        GEO      F         1941.0   

    Title  Standard_Rating  Rapid_rating  Blitz_rating Inactive_flag  
0      GM             2675        2646.0        2736.0            wi  
1      GM             2658        2621.0        2601.0            wi  
2      GM             2586        2483.0        2483.0            wi  
3      GM             2582        2502.0        2441.0            wi  
4      GM             2577        2502.0        2441.0            wi  
5      GM             2574        2502.0        2441.0            wi  
6      GM             2560        2610.0        2536.0            wi  
7      GM             2546        2521.0        2608.0            wi  
8      GM             2544        2506.0        2326.0            wi  
9      GM             2538        2394.0        2435.0            wi  
10     GM             2535        2533.0        2505.0            wi  
11     GM             2524        2447.0        2391.0            wi  
12     GM             2515        2450.0        2422.0            wi  
13     GM             2510        2501.0        2510.0            wi  
14     GM             2505        2527.0        2530

In [91]:
# Total number of Women GrandMasters in the world
len(df[df['Title'] == 'GM'])

37

In [92]:
print('Currently, the total number of Women Chess Grandmasters in the World are 37')

Currently, the total number of Women Chess Grandmasters in the World are 37


# The number of GrandMasters in the different Federations

In [93]:
countries_Gm = df[df.Title=='GM'].Federation.value_counts().reset_index().rename(columns={'index':'Country', 'Federation':'Total GMs'})[:10]
countries_Gm

Country  Total GMs
0     CHN          7
1     RUS          6
2     GEO          5
3     UKR          4
4     HUN          3
5     IND          2
6     NED          1
7     SCO          1
8     QAT          1
9     BUL          1

# Extra: Show those with Standard Rating above 2500 and less than 2500

In [94]:
df[df['Standard_Rating'] > 2500]

Fide id                     Name Federation Gender  Year_of_birth Title  \
0     700070            Polgar, Judit        HUN      F         1976.0    GM   
1    8602980               Hou, Yifan        CHN      F         1994.0    GM   
2    5008123            Koneru, Humpy        IND      F         1987.0    GM   
3    4147103  Goryachkina, Aleksandra        RUS      F         1998.0    GM   
4     700088            Polgar, Susan        HUN      F         1969.0    GM   
5    8600147                 Xie, Jun        CHN      F         1970.0    GM   
6    8603006               Ju, Wenjun        CHN      F         1991.0    GM   
7   14109336          Lagno, Kateryna        RUS      F         1989.0    GM   
8   14114550         Muzychuk, Mariya        UKR      F         1992.0    GM   
9   12801259       Cmilyte, Viktorija        LTU      F         1983.0    GM   
10  14111330           Muzychuk, Anna        UKR      F         1990.0    GM   
11  13601903          Dzagnidze, Nana        GEO      F         1987.0    GM   
12   5015197       Harika, Dronavalli        IND      F         1991.0    GM   
13   8603642             Tan, Zhongyi        CHN      F         1991.0    GM   
14   8605114             Lei, Tingjie        CHN      F         1997.0    GM   

    Standard_Rating  Rapid_rating  Blitz_rating Inactive_flag  
0              2675        2646.0        2736.0            wi  
1              2658        2621.0        2601.0            wi  
2              2586        2483.0        2483.0            wi  
3              2582        2502.0        2441.0            wi  
4              2577        2502.0        2441.0            wi  
5              2574        2502.0        2441.0            wi  
6              2560        2610.0        2536.0            wi  
7              2546        2521.0        2608.0            wi  
8              2544        2506.0        2326.0            wi  
9              2538        2394.0        2435.0            wi  
10             2535        2533.0        2505.0            wi  
11             2524        2447.0        2391.0            wi  
12             2515        2450.0        2422.0            wi  
13             2510        2501.0        2510.0            wi  
14             2505        2527.0        2530.0            wi

In [95]:
df[df['Standard_Rating'] < 2500]

Fide id                        Name Federation Gender  Year_of_birth  \
17     4198026         Kashlinskaya, Alina        RUS      F         1993.0   
18    12512214  Khademalsharieh, Sarasadat        IRI      F         1997.0   
19     8602727                 Ruan, Lufei        CHN      F         1987.0   
20     8601283                   Zhao, Xue        CHN      F         1985.0   
21     8600430                   Wang, Lei        CHN      F         1975.0   
...        ...                         ...        ...    ...            ...   
8548   3302288           Reinkens, Natalia        BOL      F         1968.0   
8549    343960           Saffova, Michaela        CZE      F         1994.0   
8550   5038294            Shetye, Siddhali        IND      F         1992.0   
8551   2072491               Trakru, Priya        USA      F         2001.0   
8552   4666399         Vorpahl, Sina Fleur        GER      F         1982.0   

     Title  Standard_Rating  Rapid_rating  Blitz_rating Inactive_flag  
17      IM             2494        2384.0        2364.0            wi  
18      IM             2494        2393.0        2431.0            wi  
19     WGM             2491        2393.0        2431.0            wi  
20      GM             2486        2442.0        2407.0            wi  
21     WGM             2484        2442.0        2407.0            wi  
...    ...              ...           ...           ...           ...  
8548   WFM             1801        1664.0        1739.0            wi  
8549   WFM             1801        1791.0        1765.0            wi  
8550   WFM             1801        1884.0        1824.0            wi  
8551   WFM             1801        1884.0        1824.0            wi  
8552   WFM             1801        1884.0        1824.0            wi  

[8536 rows x 10 columns]